From 1b3b3d13a3dd87a525076d42d3455d0e03c6bcb9 Mon Sep 17 00:00:00 2001
From: Franck LENORMAND <franck.lenormand@nxp.com>
Date: Thu, 13 Sep 2018 19:04:03 +0200
Subject: [PATCH 4722/5242] MLK-19365: drivers: crypto: caamalg: ecb(aes) does
 not use IV

commit  d87e6ae62d5b11bb8d736a908eaebd175a906764 from
https://source.codeaurora.org/external/imx/linux-imx.git

The crypto template lrw(crypto/lrw.c) and xts(crypto/xts.c) require
an ecb(aes) algo as base which ivsize must be zero as ecb(aes)
doesn't need an IV.

The patch 84f5e22194 "MA-9807: Fix ecb(aes) use without an IV"
add support in caamalg for ecb(aes) without iv.

The ecb(aes) implementation of the CAAM declare an ivsize which is
against specification. So remove it to be usable with cryto templates.

Signed-off-by: Franck LENORMAND <franck.lenormand@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/crypto/caam/caamalg.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/crypto/caam/caamalg.c b/drivers/crypto/caam/caamalg.c
index 5c1b903..b1aaa50 100644
--- a/drivers/crypto/caam/caamalg.c
+++ b/drivers/crypto/caam/caamalg.c
@@ -1929,7 +1929,7 @@ struct caam_alg_template {
 			.geniv = "eseqiv",
 			.min_keysize = AES_MIN_KEY_SIZE,
 			.max_keysize = AES_MAX_KEY_SIZE,
-			.ivsize = AES_BLOCK_SIZE,
+			.ivsize = 0,
 			},
 		.class1_alg_type = OP_ALG_ALGSEL_AES | OP_ALG_AAI_ECB,
 	},
-- 
1.7.9.5

